contract/src/api.rs

```diff
@@ -176,10 +184,25 @@ impl Contract {
             // First key receives all the deposit.
             attached_balance = 0;
         }
+
         if attached_balance > 0 {
             env::panic_str("The attached deposit could not be added to any account");
         }
-        SetReturnType { block_height: U64(near_sdk::env::block_height()) }
+
+        if options.callback_receiver_id.is_some() {
+            let callback_receiver_id = options.callback_receiver_id.unwrap();
```

Comment:
Let's avoid unnecessary unwrap:

```suggestion
        if let Some(callback_receiver_id) = options.callback_receiver_id {
```